docs/protocol.md

```diff
@@ -0,0 +1,63 @@
+# Protocol
+
+## Wire Message
+
+```
+COMMAND data
+```
+
+The data portion is what the command handlers work on.
+
+### Generic Response Messages
+
+#### OK
+```
+200 ok
+```
+Ok can be replaced by a message
+
+#### ERR
+```
+code string
+```
+code is an integer number for the given code. A code of 0 means this is a custom error message.
+
+### USE
+#### UseRequest
+```
+dbName
+```
+The name of the database to switch too.
+
+#### UseResponse
+Generic Ok
+
+### QUERY
+#### QueryRequest
+```
+string
+```
+Query is just a string extracted from the data segment.
+
+#### QueryResponse
+TODO
+
+### APPEND
+#### AppendRequest
+```
+topic data
+```
+Append is sent in two parts. Topic is the path for this data item.
```

Comment:
Since topics always begin with a leading /, how bad of an idea would it be to assume that if data doesn't look like it begins with a topic, we default to "/"?

Reply:
I had thought about doing optional topics in the PR, but decided to hold off. I think I want to add the length based message shape to the AppendRequest as well to make it very easy to do. This will require the REPL client to parse the plaintext into the proper message, but that change will be better for an API.